File: src/main/scala/com/github/kazuhito_m/twitterbattler/primitive/domain/model/battle/BattleRepository.scala
package com.github.kazuhito_m.twitterbattler.primitive.domain.model.battle

import com.github.kazuhito_m.twitterbattler.primitive.domain.model.BattleScene

trait BattleRepository {

  /**
    * BattleSceneオブジェクトを保存する。すでにある場合は上書きする。
    */
  def saveBattleScene(playerTwitterId: String, battleScene: BattleScene): Unit

  /**
    * BattleSceneオブジェクトをIDで削除する。
    */
  def deleteBattleScene(playerTwitterId: String): Unit

  /**
    * BattleSceneオブジェクトが存在するかを真偽値で返す。
    */
  def isExistsBattleScene(playerTwitterId: String): Boolean

  /**
    * 現在の「指定ユーザの戦闘画面シーンID」を返す。
    */
  def getBattleScene(playerTwitterId: String): BattleScene

  def createBattle(playerTwitterId: String): Battle

  def getBattle(playerTwitterId: String): Battle

}

File: server/src/main/java/com/github/kazuhito_m/twitterbattler/primitive/application/repository/BattleRepository.java
package com.github.kazuhito_m.twitterbattler.primitive.application.repository;

import com.github.kazuhito_m.twitterbattler.primitive.domain.model.battle.Battle;
import com.github.kazuhito_m.twitterbattler.primitive.domain.model.battle.BattleScene;

public interface BattleRepository {
    /**
     * BattleSceneオブジェクトを保存する。すでにある場合は上書きする。
     */
    void saveBattleScene(String playerTwitterId, BattleScene battleScene);

    /**
     * BattleSceneオブジェクトをIDで削除する。
     */
    void deleteBattleScene(String playerTwitterId);

    /**
     * BattleSceneオブジェクトが存在するかを真偽値で返す。
     */
    boolean isExistsBattleScene(String playerTwitterId);

    /**
     * 現在の「指定ユーザの戦闘画面シーンID」を返す。
     */
    BattleScene getBattleScene(String playerTwitterId);

    Battle createBattle(String playerTwitterId);

    void registerBattle(String playerTwitterId, Battle battle);

    Battle getBattle(String playerTwitterId);
}
